server/schedulers/balance_region.go

```diff
@@ -201,7 +205,7 @@ func (s *balanceRegionScheduler) Schedule(cluster opt.Cluster) []*operator.Opera
 				continue
 			}
 
-			if op := s.transferPeer(plan); op != nil {
+			if op := s.transferPeer(plan, stores[index:]); op != nil {
```

Comment:
Use sorted `sourceStores` and `targetStores` and `indexSource` and `indexTarget`.